Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Registration.hs

```diff
@@ -329,9 +330,11 @@ registerFleetOwner req = do
       Just NORMAL_FLEET -> FLEET_OWNER
       Nothing -> FLEET_OWNER
 
-buildFleetOwner :: (EncFlow m r) => FleetRegisterReq -> Id DRole.Role -> DRole.DashboardAccessType -> m PT.Person
-buildFleetOwner req roleId dashboardAccessType = do
-  pid <- generateGUID
+buildFleetOwner :: (EncFlow m r) => FleetRegisterReq -> Maybe Text -> Id DRole.Role -> DRole.DashboardAccessType -> m PT.Person
+buildFleetOwner req mbPersonId roleId dashboardAccessType = do
+  pid <- case mbPersonId of
+    Just personId -> return $ Id personId
+    Nothing -> generateGUID
```

Comment:
It will always be Nothing, as you are passing Nothing from top. Why is this logic then?

Reply:
no like , this is for backward compatibilty  case that is i  have created a new api fleet/register this we will be using now , since we have to keep personId same for both (bpp, dashboard ) so first creating person in bpp side , then creating it in dashboard side.